### PR TITLE
Online events

### PR DESCRIPTION
**Pull request by nmorduch**
- Events can have a link to a webcast
- Events can be marked as online only and the address is optional

### CODE REVIEW

File: newamericadotorg/assets/react/components/ContentCards.js [hunk 1]
```diff
@@ -44,7 +44,7 @@ export const EventItem = ({ event }) => (
         </h4>
         <h6 className="margin-10">{event.story_excerpt}</h6>
         <h6 className="caption margin-0">
-          {event.city}, {event.state}
+          {event.online_only ? 'Online Only' : [event.city, event.state ].filter(e => e).join(', ')}
```
Comment (nmorduch) [inline newamericadotorg/assets/react/components/ContentCards.js]: This is probably a silly way to do this but 🤷‍♀

File: newamericadotorg/templates/components/post_main.html [hunk 2]
```diff
@@ -32,7 +32,25 @@ <h6 class="paragraph margin-top-5 margin-bottom-0">{% generate_dateline page %}<
               <div class="event-details col-lg-12 col-6">
                 <h4 class="margin-0">Where</h4>
                 <h6 class="paragraph margin-top-5 margin-bottom-0">
-                  {{ page.host_organization }}<br/>{{ page.street_address }} {{ page.city }}, {{ page.state }} {{ page.zipcode }}
+                  <div>{{ page.host_organization }}</div>
```
Comment (nmorduch) [inline newamericadotorg/templates/components/post_main.html]: I guess I could have moved this whole section to a partial but 🤷‍♀

File: event/models.py [hunk 3]
```diff
@@ -40,11 +41,14 @@ class Event(Post):
         blank=True,
         null=True
     )
-    street_address = models.TextField(default='740 15th St NW #900')
-    city = models.TextField(default='Washington')
-    state = models.TextField(default='D.C.')
-    zipcode = models.TextField(default='20005')
-    rsvp_link = models.URLField(blank=True)
+    street_address = models.TextField(default='740 15th St NW #900', blank=True, null=True)
```
Comment (kaedroho) [inline event/models.py]: Text/Char/URL fields only need `blank=True` to make not required

File: newamericadotorg/assets/react/components/ContentCards.js (hunk 1 above)
Comment (kaedroho) [reply newamericadotorg/assets/react/components/ContentCards.js]: Looks fine to me!

File: newamericadotorg/templates/components/post_heading_image.html [hunk 4]
```diff
@@ -25,7 +25,15 @@ <h6 class="paragraph">{% generate_dateline page %}</h6>
         <div class="event-details col-6">
           <h4 class="margin-0">Where</h4>
           <h6 class="paragraph">
-            {{ page.host_organization }}<br/>{{ page.street_address }} {{ page.city }}, {{ page.state }} {{ page.zipcode }}
+            <div>{{ page.host_organization }}</div>
+            <div>
+              {% include 'components/_address_text.html' with address_text=page %}
+            </div>
+            {% if page.webcast_url %}
+              <a class="button--text with-caret--right" href="{{page.webcast_url}}" target="_blank">
```
Comment (kaedroho) [inline newamericadotorg/templates/components/post_heading_image.html]: We should add ``rel="noopener noreferrer"`` to this tag. (see https://www.jitbit.com/alexblog/256-targetblank---the-most-underestimated-vulnerability-ever/)

Comment (nmorduch) [reply newamericadotorg/templates/components/post_heading_image.html]: Ooh yes thank you. We should do that globally.

File: event/models.py (hunk 3 above)
Comment (nmorduch) [reply event/models.py]: Thank you! Good to know :)

File: event/models.py [hunk 5]
```diff
@@ -38,13 +39,15 @@ class Event(Post):
...
-    city = models.TextField(default='Washington')
-    state = models.TextField(default='D.C.')
-    zipcode = models.TextField(default='20005')
-    rsvp_link = models.URLField(blank=True)
+    street_address = models.TextField(default='740 15th St NW #900', blank=True)
+    city = models.TextField(default='Washington', blank=True)
+    state = models.TextField(default='D.C.', blank=True)
+    zipcode = models.TextField(default='20005', blank=True)
+    online_only = models.BooleanField(default=False, help_text="Checking this will mean the address is not shown")
+    webcast_url = models.URLField(blank=True, null=True)
```
Comment (kaedroho) [inline event/models.py]: No need for `null=True` on `URLField` either (these are just `TextField` with a bit of extra validation).

It'll just use a blank string to represent emptiness.

**Pull request merged**